projects/solar/test/test_fan_control_solar.c

```diff
@@ -0,0 +1,234 @@
+#include "fan_control_solar.h"
+
+#include <stdbool.h>
+
+#include "data_store.h"
+#include "event_queue.h"
+#include "exported_enums.h"
+#include "fault_handler.h"
+#include "fault_monitor.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "max6643_fan_controller.h"
+#include "ms_test_helpers.h"
+#include "solar_boards.h"
+#include "solar_events.h"
+#include "spv1020_mppt.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+// Any defines here
+
+// Declare static test variables
+static GpioAddress s_overtemp_addr;
+static GpioAddress s_fan_fail_addr;
+static GpioAddress s_full_speed_addr;
+static uint16_t s_full_speed_temp_threshold;
+static SolarMpptCount s_mppt_count;
+
+static GpioState s_gpio_state;
+static Event s_e;
+
+static GpioAddress s_dummy;
+
+static void TEST_MOCK(prv_fanfail_callback)(const GpioAddress *address, void *context) {
+  LOG_WARN("fan_control detected fanfail, raising fault event");
+  fault_handler_raise_fault(EE_SOLAR_FAULT_FAN_FAIL, 0);
+}
+
+void setup_test(void) {
+  event_queue_init();
+  gpio_init();
+  gpio_it_init();
+  data_store_init();
```

Comment:
You also need to call `interrupt_init` (from interrupt.h) since gpio_it depends on it